React/Modules/RCTTiming.m

```diff
@@ -163,6 +176,11 @@ - (void)startTimers
   }
 }
 
+- (BOOL)hasPendingTimers
+{
+  return _sendIdleEvents || _timers.count > 0;
+}
+
 - (void)didUpdateFrame:(__unused RCTFrameUpdate *)update
```

Comment:
So this will get called at the beginning I believe of every frame, I don't think this does what you want it to do in terms of sending idle events.